### PR TITLE
Add support for using NonEmptyList as get paramter

### PR DESCRIPTION
**Pull request by strokyl**
Today, I was writing an endpoint with NonEmptyList as a result for a get parameter.
I had to add this line too make it work.
Would you be interested in providing PlainCodec for NonEmpty collection of cats here ?
If yes is it the correct way to do it ?
And if you want I can write the code for other NonEmpty collecttion.
Also I am not quite sure, how to test codec

### CODE REVIEW

**Comment by adamw** [comment]
Sure, sounds good.

That's the correct place to add the codec. One minor difference is that instead of `map`, I'd use `mapDecode` and report a missing element if the given list is empty (using `DecodeResult.Missing`)

A good way to test codecs could be as e.g. in `FormCodecDerivationTest`, so you encode a decode a value and test that it's the same. Could be even a good property test :) Plus some tests for empty lists, which should prodce a failed decode result.

File: cats/src/test/scala/sttp/tapir/codec/cats/TapirCodecCatsTest.scala [hunk 1]
```diff
@@ -30,4 +36,64 @@ class TapirCodecCatsTest extends FlatSpec with Matchers {
...
+
+  implicit def arbitraryNonEmptyChain[T: Arbitrary]: Arbitrary[NonEmptyChain[T]] = Arbitrary(
+    Gen.nonEmptyListOf(implicitly[Arbitrary[T]].arbitrary).map(NonEmptyChain.fromSeq(_).get)
+  )
+
+  implicit def arbitraryNonEmptySet[T: Arbitrary: Ordering]: Arbitrary[NonEmptySet[T]] = Arbitrary(
+    Gen.nonEmptyBuildableOf[SortedSet[T], T](implicitly[Arbitrary[T]].arbitrary).map(NonEmptySet.fromSet(_).get)
+  )
+
+  "Provided PlainText coder for non empty list" should "correctly serialize a non empty list" in {
```
Comment (strokyl) [inline cats/src/test/scala/sttp/tapir/codec/cats/TapirCodecCatsTest.scala]: I know I could have factorize test logic. But when its not too much of a burden I like to keep test simple.

File: core/src/main/scala/sttp/tapir/DecodeResult.scala [hunk 2]
```diff
@@ -29,4 +29,6 @@ object DecodeResult {
         }
     }
   }
+
+  def apply[T](o: Option[T]): DecodeResult[T] = o.map(Value(_)).getOrElse(Missing)
```
Comment (strokyl) [inline core/src/main/scala/sttp/tapir/DecodeResult.scala]: Tell me if it is ok, but I think transforming an Option into a DecodeResult will often make sens.
But maybe apply is too much maybe I should have name it "fromOption".
WDYT?

Comment (adamw) [reply core/src/main/scala/sttp/tapir/DecodeResult.scala]: Yes I think `fromOption` communicates the intent more clearly

File: cats/src/test/scala/sttp/tapir/codec/cats/TapirCodecCatsTest.scala (hunk 1 above)
Comment (adamw) [reply cats/src/test/scala/sttp/tapir/codec/cats/TapirCodecCatsTest.scala]: looks good :)

File: core/src/main/scala/sttp/tapir/DecodeResult.scala (hunk 2 above)
Comment (strokyl) [reply core/src/main/scala/sttp/tapir/DecodeResult.scala]: Done

**Comment by strokyl** [comment]
Great! Thank you a lot, could you tell me when it will be released?

**Comment by adamw** [comment]
Trying to make a release but sonatype seems to be having problems (again ;) )

**Comment by adamw** [comment]
Released in 0.12.18, thanks!

**Pull request merged**